ngx-tools/src/utilities/object-deep-parse.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { objectDeepParse } from './object-deep-parse';
+
+describe(`objectDeepParse`, () => {
+  const stringInput = 'name';
+  const objectInput = {foo: {bar: {baz: 'xyz'}}};
+
+  test(`when string input, output same string value`, () => {
```

Comment:
Let's flip these sentences around to match the repo's `should ...` format. (ie `should output same string value when string input` etc

Or if having the preface helps with clarity, in a few places I've wrapped another `describe` around the test:

```typescript
describe('objectDeepParse')
  describe('when passed a string')
    test('should output the same string value')
  describe('when passed an object')
    ...
```